flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -606,18 +649,24 @@ private static long calculateRepeatIntervalMilliseconds(NotificationDetails noti
     return repeatInterval;
   }
 
+  // Note: This will now allow duplicate notifications on ID which would be overwritten before.
   private static void saveScheduledNotification(
       Context context, NotificationDetails notificationDetails) {
-    ArrayList<NotificationDetails> scheduledNotifications = loadScheduledNotifications(context);
-    ArrayList<NotificationDetails> scheduledNotificationsToSave = new ArrayList<>();
-    for (NotificationDetails scheduledNotification : scheduledNotifications) {
-      if (scheduledNotification.id.equals(notificationDetails.id)) {
-        continue;
-      }
-      scheduledNotificationsToSave.add(scheduledNotification);
+    SharedPreferences sharedPreferences =
+        context.getSharedPreferences(SCHEDULED_NOTIFICATIONS_FILE, Context.MODE_PRIVATE);
+    Set<String> jsonNotifications = getScheduledNotificationsJsonSet(sharedPreferences);
+    if (jsonNotifications == null) {
+      jsonNotifications = new HashSet<String>();
     }
-    scheduledNotificationsToSave.add(notificationDetails);
-    saveScheduledNotifications(context, scheduledNotificationsToSave);
+
+    // Mutations on the Set from getStringSet() are not allowed!
+    Set<String> jsonNotificationsToSave = new HashSet<String>(jsonNotifications);
+
+    Gson gson = buildGson();
+    jsonNotificationsToSave.add(gson.toJson(notificationDetails));
```

Comment:
notifications should have unique ids not unique json strings

Reply:
I ignored this explicitly, although we can solve this by using a HashMap instead of ArrayList in `loadScheduledNotifications`.